### DISCUSSION TITLE
Deepchecks on distributed computing frameworks: Dask and Polars

### DISCUSSION THREAD

**Discussion by bgalvao** [deepchecks/deepchecks#1701] [Ideas]
Greetings! This project is amazing and I'm already experimenting writing tests for a CI/CD pipeline.

I am starting to question, what happens if the dataset is too large for Pandas? Would it be seamless to switch to [Dask](https://docs.dask.org/en/stable/dataframe.html#:~:text=A%20Dask%20DataFrame%20is%20a,different%20machines%20in%20a%20cluster.) or [Polars](https://www.pola.rs/)?

I would love to work on a Pull Request if it comes to be necessary 😁

**Comment by noamzbr** [comment]
Hi @bgalvao, very good question! 

Many of the checks in the package don't really need to run on all samples, so the first thing they do is sample the pandas DataFrame contained in the Dataset. I expect that will work with Dask, but of course it's better to test first hand. Other checks (mainly those that test data integrity, or compare datasets) use the whole data and can fail in case of a too-large dataset.

It indeed may be possible to seamlessly switch from pandas to Dask (or [Modin](https://github.com/modin-project/modin), for the option to use [Ray](https://docs.ray.io/en/latest/data/dataset.html) or Dask with the same api). 
If you got the time, I'd be very glad if you can try to use our checks with Dask! Specifically, the following checks may prove problematic and are worth testing specifically:

- ConflictingLabels
- SpecialCharacters
- StringLengthOutOfBounds
- DateTrainTestLeakageDuplicates

Please let us know if that's something you're willing to tackle. If so, we can open an issue from this discussion.

**Reply by bgalvao** [reply]
For now, I definitely want to test with Dask and Polars 😃 

It seems to me that I have to remove the specific [`pd.DataFrame` cast](https://github.com/deepchecks/deepchecks/blob/d211cf1951c5d3a6bcaf6c4e2e1ada2f3b2e7c8b/deepchecks/tabular/dataset.py#L124) and add Dask and Polars dataframe and series types to the [`isinstance` checks](https://github.com/deepchecks/deepchecks/blob/d211cf1951c5d3a6bcaf6c4e2e1ada2f3b2e7c8b/deepchecks/tabular/dataset.py#L129) in your codebase in order to let the deepchecks run.

**Reply by noamzbr** [reply]
Good catch! didn't think about it. If you're modifying the code so that it'll work with Dask, we'd appreciate if you'd consider opening a PR with these changes so that it'll be supported in the package.

**Reply by royinblr** [reply]
Dear All, We are also considering to replace pandas for parallel processing. It will be helpful if anyone has already tried with any other parallel processing such as DASK, PySpark, Ray or any share suggestion on the same. It would be appreciated.

**Reply by bgalvao** [reply]
The big question for me is how would an abstract class / interface be implemented to abstract the engine.

It won't be just the initialization of some test suite for example. But also, all of the other functions that implement the tests will need to have engine-specific code.

Still short on time for implementing this, but still think it is an interesting discussion to have.
